core/domain/skill_domain_test.py

```diff
@@ -111,6 +114,29 @@ def test_misconception_validation(self):
         self.skill.misconceptions = ''
         self._assert_validation_error('Expected misconceptions to be a list')
 
+    def test_misconception_validation_with_invalid_html_in_notes(self):
+        self.skill.misconceptions[0].notes = '<a>Test</a>'
+        self._assert_validation_error(
+            'Invalid html: <a>Test</a> for rte with '
+            'invalid tags and strings: .+')
```

Comment:
The output is formatted as a dict in the error message, so it does not match the regex string due to the formatting. I get an error here as:
```
"Invalid html: <a>Test</a> for rte with invalid tags and strings: {'invalidTags': [u'a'], 'strings': ['<a>Test</a>']}" does not match "Invalid html: <a>Test</a> for rte with invalid tags and strings: {'invalidTags': [u'a'], 'strings': ['<a>Test</a>']}"
```

Reply:
As discussed, this needs a little more digging to figure out the actual cause (but if you find that you're digging too much, let's keep it as regex and move on). Would be better to have the whole thing though since that's kind of the important part.

One thought -- maybe make sure it's not some kind of type issue (str vs unicode or something).

Reply:
This is fixed now, yeah!! I used your suggestion to check the output by removing some characters and find out where the mismatch was occurring. The issue was due to square brackets in the output which were being used as an operator for regex to group the set of characters.

To be clear, the square brackets here `[u'a']` are being treated as a group of characters which can occur in the regex. Adding a slash to treat them as literal instead of group solves the problem. I replaced `[u'a']` with `\\[u'a']`

Reply:
That makes sense -- @ankita240796, well done on figuring this out! :D